prov/cxi/src/cxip_rxc.c

```diff
@@ -402,6 +402,13 @@ struct cxip_rxc *cxip_rxc_calloc(struct cxip_ep_obj *ep_obj, void *context)
 {
 	struct cxip_rxc *rxc = NULL;
 
+	/* 
+	 * It's possible the owner provider decides to turn off
+	 * hardware offload in cxi. If that happens we need to update the
+	 * rx_match_mode.
+	 */
+	cxip_set_env_rx_match_mode();
```

Comment:
If a future patch, we need to make this a domain fi_control() arg to avoid this. This is what we have done for other env vars.